samtranslator/model/connector/connector.py

```diff
@@ -174,30 +179,43 @@ def get_resource_reference(
     )
 
 
+def _get_events_rule_role(
+    connecting_obj_id: Optional[str], connecting_obj_arn: Optional[Any], properties: Dict[str, Any]
+) -> Optional[Any]:
+    for target in properties.get("Targets", []):
+        target_arn = target.get("Arn")
+        target_logical_id = get_logical_id_from_intrinsic(target_arn)
+        if (target_logical_id and target_logical_id == connecting_obj_id) or (
+            connecting_obj_arn and target_arn == connecting_obj_arn
+        ):
+            return target.get("RoleArn")
+    return None
+
+
 def _get_resource_role_property(
-    connecting_obj_id: Optional[str], connecting_obj_arn: Optional[Any], resource_type: str, properties: Dict[str, Any]
+    connecting_obj_id: Optional[str],
+    connecting_obj_arn: Optional[Any],
+    cfn_resource_properties: Dict[str, Any],
+    properties: Dict[str, Any],
 ) -> Any:
-    if resource_type == "AWS::Lambda::Function":
-        return properties.get("Role")
-    if resource_type == "AWS::StepFunctions::StateMachine":
-        return properties.get("RoleArn")
-    if resource_type == "AWS::AppSync::DataSource":
-        return properties.get("ServiceRoleArn")
-    if resource_type == "AWS::Events::Rule":
-        for target in properties.get("Targets", []):
-            target_arn = target.get("Arn")
-            target_logical_id = get_logical_id_from_intrinsic(target_arn)
-            if (target_logical_id and target_logical_id == connecting_obj_id) or (
-                connecting_obj_arn and target_arn == connecting_obj_arn
-            ):
-                return target.get("RoleArn")
+    role_property = cfn_resource_properties.get("Inputs", {}).get("Role")
+
+    if isinstance(role_property, str):
+        return properties.get(role_property)
+
+    if isinstance(role_property, dict) and role_property.get("Function") == "GetEventsRuleRole":
```

Comment:
So this only supports `Function` on `Role`? Is there a more generic approach we can take (that doesn't make the code too complex) so we don't need to change multiple places in code later if we support more `Function`s?

Reply:
I don't see how other properties would every need a custom function.

Reply:
Yeah I see only input we have is for role. And in any case we can adjust the code later; any issues should be caught by transform tests when we add new connectors.